### PR TITLE
Crystal 0.36.0 support

### LINKED ISSUE #3: 0.36.0 - compiler error: macro []?(key) invalid macro name
Currently 0.36.0 is throwing this error for secrets-env as a shard

```crystal
In lib/secrets-env/src/secrets-env.cr:21:12

 21 | macro []?(key)
               ^
Error: invalid macro name
```

### PR DESCRIPTION
**Pull request by kimburgess**
Adds (partial) support for crystal 0.36.0 along with some better test coverage.

Core functionality remains, however static resolution of keys accessed via a non-strict lookup (`[]?`) in not possible due to stricter macro name checks.

### CODE REVIEW

**Comment by stakach** [comment]
nice, it looks like your pull to re-introduce `[]?` has legs, which is good
hopefully they release 0.36.1 soon

File: src/secrets-env.cr [hunk 1]
```diff
@@ -4,23 +4,52 @@ require "path"
...
+      # Override `.[]?` to enable compile-time resolution or accessed keys.
+      #
+      # Maintains the behaviour of the method of the same name.
+      macro []?(key)
+        {{ ACCESSED << key if key.is_a? StringLiteral && !ACCESSED.includes? key }}
+        ENV.fetch({{ key }}, nil)
+      end
+    {% end %}
+  {% else %}
+    STDERR.puts "Warning: unsupported Crystal version, ENV.accessed will only provide partial results"
```
Comment (caspiano) [inline src/secrets-env.cr]: Should we put this log here, or set a crystal version pin in the shard.yml?
It's going to be partial but if (big if) all env accesses happen at boot and `fetch` is still overridden, it should still sort of work?

Comment (caspiano) [reply src/secrets-env.cr]: Either way, hopefully we get a patch soon 😬

Comment (stakach) [reply src/secrets-env.cr]: I think this warning is good enough for this version of crystal, we can re-evaluate if `[]?` macros don't make it into the next version

**Comment by caspiano** [comment]
Version bump and LGTM

File: src/secrets-env.cr (hunk 1 above)
Comment (kimburgess) [reply src/secrets-env.cr]: I've narrowed the impact / visibility of this in 181c9ea. Functionality should be there for _most_ cases, however if there is an attempt to do a static only lookup on 0.36.0 there's some guard rails in place.

**Comment by caspiano** [comment]
I'm getting this type error 

```
 47 | def self.accessed(static_only = false) : Enumerable(String)
                                               ^
Error: method ENV.accessed must return Enumerable(String) but it is returning (Array(String) | Tuple())
```

**Comment by kimburgess** [comment]
Nice catch! Spec interacts with the `ENV` module so I don't think it's possible to add a test case to cover that scenario. dc5691f should address the cause though.

**Review by stakach — APPROVED** [review]
LGTM

**Comment by caspiano** [comment]
One more

```
 28 | ENV.accessed.sort.each &->puts(String)
                   ^---
Error: undefined method 'sort' for StaticArray(String, 0) (compile-time type is (Array(String) | StaticArray(String, 0)))
```

**Pull request merged**